src/coreclr/interpreter/stackmap.cpp

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "gcinfoencoder.h" // for GcSlotFlags
+
+// HACK: debugreturn.h (included by gcinfoencoder.h) breaks constexpr
+#if defined(debug_instrumented_return) || defined(_DEBUGRETURN_H_)
+#undef return
+#endif // debug_instrumented_return
+
+#include "interpreter.h"
+#include "stackmap.h"
+
+extern "C" {
+    #include "../../native/containers/dn-simdhash.h"
+    #include "../../native/containers/dn-simdhash-specializations.h"
+
+    void assertAbort(const char* why, const char* file, unsigned line);
+
+    void
+    dn_simdhash_assert_fail (const char* file, int line, const char* condition);
+
+    void
+    dn_simdhash_assert_fail (const char* file, int line, const char* condition) {
+        assertAbort(condition, file, line);
+    }
+}
+
+thread_local dn_simdhash_ptr_ptr_t *t_sharedStackMapLookup = nullptr;
+
+InterpreterStackMap* GetInterpreterStackMap(ICorJitInfo* jitInfo, CORINFO_CLASS_HANDLE classHandle)
+{
+    InterpreterStackMap* result = nullptr;
+    if (!t_sharedStackMapLookup)
+        t_sharedStackMapLookup = dn_simdhash_ptr_ptr_new(0, nullptr);
+    if (!dn_simdhash_ptr_ptr_try_get_value(t_sharedStackMapLookup, classHandle, (void **)&result))
+    {
+        result = new InterpreterStackMap(jitInfo, classHandle);
+        dn_simdhash_ptr_ptr_try_add(t_sharedStackMapLookup, classHandle, result);
+    }
+    return result;
+}
+
+void InterpreterStackMap::PopulateStackMap(ICorJitInfo* jitInfo, CORINFO_CLASS_HANDLE classHandle, uint32_t offset)
+{
+    unsigned size = jitInfo->getClassSize(classHandle);
+    unsigned maxGcPtrs = (size + sizeof(void *) - 1) / sizeof(void *);
+    if (maxGcPtrs < 1)
+        return;
+
+    uint8_t *gcPtrs = (uint8_t *)alloca(maxGcPtrs);
+    unsigned numGcPtrs = jitInfo->getClassGClayout(classHandle, gcPtrs),
+        newCapacity = m_slotCount + numGcPtrs;
+
+    m_slots = (InterpreterStackMapSlot *)realloc(m_slots, sizeof(InterpreterStackMapSlot) * newCapacity);
```

Comment:
Seems to me that for every class we create a stack map that is populated here and `m_slots` is initialized only once. I don't understand why we are using realloc ? Could we use malloc just to avoid confusion ?